polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala

```diff
@@ -72,16 +67,43 @@ class PythonInterpreter(val symbolTable: RuntimeSymbolTable) extends LanguageKer
     // TODO: Instead of doing this, can we somehow just make the symbol table a Python scope?
     symbolTable.subscribe() {
       value => withJep {
-        if (!value.source.contains(this)) {
-          jep.set(value.name.toString, value.value)
+        value.value match {
```

Comment:
It looks like you took out the source filter here, but also aren't using the `subscriber` option you introduced?

Reply:
Oops, thanks for the catch, missed it!

Reply:
Fixed!

Reply:
I'm not sure I get the motivation for the `Option[Any]` subscriber thing, let's discuss (not that it matters since `RuntimeSymbolTable` has a price on its head...)

Reply:
The motivation is just because I thought it seemed weird that a called would subscribe to its own events (is there any case that would be useful?). I figured that subscription code would commonly end up looking like 

```scala
symbolTable.subscribe() { value => if (!value.source.contains(this)) { ... } else { IO.unit }}
```

and I just wanted to save callers the trouble. 

Definitely open to alternatives, esp. ways to get rid of the `Option[Any]` which I agree is ugly. Also open to reverting to the old way too 😄